apps/passport-client/components/screens/ImportBackupScreen.tsx

```diff
@@ -0,0 +1,386 @@
+import { EmailPCDTypeName } from "@pcd/email-pcd";
+import { deserializeStorage } from "@pcd/passport-interface";
+import { Spacer, styled } from "@pcd/passport-ui";
+import { PCDCollection } from "@pcd/pcd-collection";
+import { PCD } from "@pcd/pcd-types";
+import { SemaphoreGroupPCDTypeName } from "@pcd/semaphore-group-pcd";
+import { SemaphoreIdentityPCDTypeName } from "@pcd/semaphore-identity-pcd";
+import { useCallback, useEffect, useState } from "react";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, usePCDCollection } from "../../src/appHooks";
+import { getPackages } from "../../src/pcdPackages";
+import { AppState } from "../../src/state";
+import { useSelector } from "../../src/subscribe";
+import { Button, H2 } from "../core";
+import { MaybeModal } from "../modals/Modal";
+import { AppContainer } from "../shared/AppContainer";
+import { ScreenNavigation } from "../shared/ScreenNavigation";
+
+export function useImportScreenData() {
+  return useSelector<AppState["importScreen"]>((s) => s.importScreen, []);
+}
+
+const NoFolderSymbol = Symbol("None");
+
+// There are four main UI states that can occur after a user selects a file
+// to import.
+type ImportState =
+  // Initial state
+  | { state: "initial" }
+  // The selected file is valid, and the user can decide whether to import it.
+  | {
+      state: "valid-file-selected";
+      // The collection parsed from the selected file
+      collection: PCDCollection;
+      // The PCD IDs referring to PCDs which are valid to import
+      mergeablePcdIds: Set<PCD["id"]>;
+      // The folders the user has selected in the UI
+      selectedFolders: Set<string | symbol>;
```

Comment:
this `| symbol` thing is interesting, I'm not sure I like it. The way I've handled the 'no folder' concept elsewhere is by canonically representing it as the empty string. I admit I haven't done that with 100% consistency. You could do the same thing here, and then present it at the UI layer as whatever you want. Having read the rest of this PR, I think it may simplify a good chunk of code.

Reply:
Oh, the empty string is much easier. Good point.